test/cases/edge.sh

```diff
@@ -0,0 +1,360 @@
+        break
+        ;;
+      "failure")
+        echo "Image compose failed"
+        exit 1
+        ;;
+      *)
+        echo "API returned unexpected image_status.status value: '$COMPOSE_STATUS'"
+        exit 1
+        ;;
+    esac
+    sleep 30
+  done
+}
+
+# Download image from aws s3
+function download_image() {
+    RESULT=$($CURLCMD -H "$HEADER" --request GET "$BASEURL/composes/$COMPOSE_ID")
+    RESULT_URL=$(get_response "$RESULT" | jq -r '.image_status.upload_status.options.url')
+    echo "Downloading image, URL is: $RESULT_URL, save it as $1"
+    $CURLCMD "$RESULT_URL" --output "$1"
+}
+
+# Test result checking
+check_result () {
+    greenprint "🎏 Checking for test result"
+    if [[ $RESULTS == 1 ]]; then
+        greenprint "💚 Success"
+    else
+        greenprint "❌ Failed"
+        after_test
+        exit 1
+    fi
+}
+
+# Wait for the ssh server up to be.
+function wait_for_ssh() {
+    SSH_STATUS=$(sudo ssh "${SSH_OPTIONS[@]}" -i "${SSH_KEY}" admin@"${1}" '/bin/bash -c "echo -n READY"')
+    if [[ $SSH_STATUS == READY ]]; then
+        echo 1
+    else
+        echo 0
+    fi
+}
+
+# Run before test begin to prepare test environment
+function before_test() {
+    # Check service status
+    READY=0
+    for RETRY in {1..10};do
+    curl --fail -H "$HEADER" "http://$ADDRESS:$PORT/ready" && {
+        READY=1
+        break
+    }
+    echo "Port $PORT is not open. Waiting...($RETRY/10)"
+    sleep 1
+    done
+
+    [ "$READY" -eq 1 ] || {
+    echo "Port $PORT is not open after retrying 10 times. Exit."
+    exit 1
+    }
+
+    # ansible is not in RHEL repositories, enable EPEL and install ansible manually.
+    sudo dnf install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm
+    sudo dnf install -y ansible
+
+    # Start libvirtd and test it.
+    greenprint "🚀 Starting libvirt daemon"
+    sudo systemctl start libvirtd
+    sudo virsh list --all > /dev/null
+
+    if ! sudo virsh net-info integration > /dev/null 2>&1; then
+        sudo virsh net-define ${IMAGE_BUILDER_TEST_DATA}/edge/integration.xml
+        sudo virsh net-start integration
+    fi
+
+    # Allow anyone in the wheel group to talk to libvirt.
+    greenprint "🚪 Allowing users in wheel group to talk to libvirt"
+    WHEEL_GROUP=wheel
+    if [[ $ID == rhel ]]; then
+        WHEEL_GROUP=adm
+    fi
+
+    sudo tee /etc/polkit-1/rules.d/50-libvirt.rules > /dev/null << EOF
+polkit.addRule(function(action, subject) {
+    if (action.id == "org.libvirt.unix.manage" &&
+        subject.isInGroup("${WHEEL_GROUP}")) {
+            return polkit.Result.YES;
+    }
+});
+EOF
+
+    # Ensure SELinux is happy with our new images.
+    greenprint "👿 Running restorecon on image directory"
+    sudo restorecon -Rv /var/lib/libvirt/images/
+
+    # Start httpd service
+    greenprint "🚀 Starting httpd daemon"
+    sudo systemctl start httpd
+
+    # Stop firewalld service because firewall will prevent edge vm installation
```

Comment:
Are you sure the firewall got installed? If not, please do not disable it to avoid issue.
From the CI result running on osbuild-composer, the openstack/rhel-8.4 image does not have firewalld installed.
Please remove firewalld configurations.

Reply:
removed firewall related code, thanks.